src/Mealz/MealBundle/Entity/ParticipantRepository.php

```diff
@@ -38,8 +39,8 @@ public function getParticipantsOnDays(
             ->orderBy('up.name', 'ASC')
             ->addOrderBy('m.dateTime', 'ASC')
             ->setParameters([
-                'startDate' => (clone $startDate)->setTime(0, 0),
-                'endDate' => (clone $endDate)->setTime(23, 59, 59),
+                'startDate' => $startDate,
```

Comment:
Agree with the clone part. I will add it again.

The removal of setTime is definitely on purpose and is what this bug is all about. The setTime manipulation led to the error that the history was wrong (please check the target process for details). I found a lot of other places where setTime was used before this database call was made (and got overridden). In general, it's a very unexpected behaviour that inside a function with a database call the parameters are modified. In my opinion, it's more intuitive, when you set the correct time from outside. I know that the setTime was set here, because you would like to request full days but for the history this was wrong. A user only pays after 12, so when the lunch has started. If you use that method the way it was before, this led to the error. We need the exact time and not the whole day. To make this even clearer, I should rename the method from getParticipantsOnDays to something without "OnDays".